frontend/src/views/admin/matching/applicant-view.tsx

```diff
@@ -0,0 +1,962 @@
+                    )}
+                </Modal.Body>
+                <Modal.Footer>
+                    <Button
+                        onClick={() => setShownApplication(null)}
+                        variant="outline-secondary"
+                    >
+                        Close
+                    </Button>
+                </Modal.Footer>
+            </Modal>
+            <Modal
+                show={showChangeHours}
+                onHide={() => setShowChangeHours(false)}
+                size="sm"
+            >
+                <Modal.Header closeButton>
+                    <Modal.Title>Update Hours</Modal.Title>
+                </Modal.Header>
+                <Modal.Body>
+                    <Form>
+                        <Form.Group className="mb-3">
+                            <Form.Control
+                                type="number"
+                                defaultValue={
+                                    applicantMatch &&
+                                    applicantMatch.hoursAssigned > 0
+                                        ? applicantMatch.hoursAssigned
+                                        : 0
+                                }
+                                onChange={(e) =>
+                                    setHoursAssigned(e.target.value)
+                                }
+                            />
+                        </Form.Group>
+                    </Form>
+                </Modal.Body>
+                <Modal.Footer>
+                    <Button
+                        onClick={() => setShowChangeHours(false)}
+                        variant="outline-secondary"
+                    >
+                        Close
+                    </Button>
+                    <Button
+                        onClick={() => {
+                            updateApplicantMatch(
+                                "staged-assigned",
+                                Number(hoursAssigned)
+                            );
+                            setShowChangeHours(false);
+                        }}
+                        variant="outline-primary"
+                    >
+                        Save
+                    </Button>
+                </Modal.Footer>
+            </Modal>
+        </div>
+    );
+}
+
+/**
+ * A button allowing one to toggle an applicant's "starred" status for the currently-sected position.
+ *
+ * @param {*} props
+ * @returns
+ */
+function ApplicantStar({
+    match,
+    updateApplicantMatch,
+    markAsUpdated,
+}: {
+    match: Match | null;
+    updateApplicantMatch: Function;
+    markAsUpdated: Function;
+}) {
+    async function _onClick(e: any) {
+        if (match) {
+            if (match.status === "applied" || match.status === "hidden") {
+                updateApplicantMatch("starred");
+            } else if (match.status === "starred") {
+                updateApplicantMatch("applied");
```

Comment:
An applicant can't be starred _and_ hidden, so you can't technically unstar a hidden applicant.
Starring a hidden applicant and then unstarring them would set them back to "applied" though, if that's what the question's getting at?

Reply:
Updated how matching works; a match will keep track of its starred and hidden statuses independent of one another now. However, in the applicant view display, there's a priority system for which category the applicant is presented under (with "assigned" > "staged-assigned" > "starred" > "hidden" > "applied").

So:
- a match that is both starred and hidden will show up under "Starred"
- if you star a hidden applicant, they will show up under "Starred"
- if you mark a starred applicant as hidden, they will still be visible under "Starred"
- if you un-star an applicant that is marked as hidden, they will now show up under "Hidden" instead of automatically dropping to "Applied"